### ISSUE TITLE
HTTPS & EPROTO error

### ISSUE THREAD

**Issue by zero-** [cgiffard/node-simplecrawler#336]
## What happened?
I tried to crawl a site that switches from http to https and I get fetch client errors. This has got something to do with https because it doesn't happen with other than https sites.

## What should have happened?
It should have fetched the page properly.

## Steps to reproduce the problem
I start to crawl `http://add.org` and it gets the homepage correctly because ADD haven't implemented https redirect to the root. All the links on the homepage are HTTPS so the following fetches are under HTTPS. Code is standard boilerplate from the examples and the error I get is `{"host":"add.org","path":"/pab/","port":443,"protocol":"https","uriPath":"/pab/","url":"https://add.org/pab/","depth":2,"referrer":"http://add.org/","fetched":true,"status":"failed","stateData":{"code":600},"id":12},{"code":"EPROTO","errno":"EPROTO","syscall":"write"}`.

I also applied fix from #288 manually because for whatever reason it doesn't seem to exist in NPM or master but all it did was that now port shows 443 correctly. Without the fix port is empty but in either case it fails.

**Comment by fredrikekelund** [comment]
Hey, thanks for filing an issue! That's kind of a special case there with http://add.org. Do I understand it correctly that even if you apply the changes from #288, it still doesn't work correctly?

simplecrawler does consider sites that use HTTP and sites that use HTTPS as different, and it shouldn't really follow links that use HTTPS from a page that was fetched over HTTP. Even though you say that they code you're using is standard boilerplate, would you mind posting a use case so that I can try and reproduce the problem?

**Comment by zero-** [comment]
Yes, I applied the changes and I see from the error that port 443 is applied correctly at least to the error message ;)

Here's the code. My URLs are domains without protocol or subdomain.

    let myCrawler = new Crawler('http://'+url);
    myCrawler.respectRobotsTxt = false
    myCrawler.useProxy = true

    let prox = _.shuffle(proxies)[0]
    myCrawler.proxyHostname = prox[0]
    myCrawler.proxyPort = prox[1]
    myCrawler.proxyUser = prox[2]
    myCrawler.proxyPass = prox[3]
    myCrawler.maxConcurrency = CONCURRENT_FETCH_THREADS
    myCrawler.maxDepth = 2
    myCrawler.scanSubdomains = false
    myCrawler.filterByDomain = true
    myCrawler.start()

**Comment by fredrikekelund** [comment]
Coming back to this now, first of all I should correct myself, because simplecrawler only makes a difference between different protocols on the same domain in certain aspects. When adding items to the queue, no distinction is made, but when it comes to robots.txt filtering, different protocol and domain name combinations are considered different. The latter aspect should be transparent however, and shouldn't be something you have to worry about.

What's more likely to be going on here is something with the proxy settings. I know it's been some time since the last activity on this issue, but if you could give me some insight into what setup you're using for your proxy, that would be great!

**Comment by geramy92** [comment]
Because I saw your discussion here: we had the same problem with our cooperate proxy, that we ended up into an ssl issue. But we resolved it by simply change the getRequestOptions function in that way, that we do not anymore replace request host, port but we set an agent with the help of the package "https-proxy-agent". Only as a small hint here.
